### PR TITLE
Update Flytekit Dockerfile

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Tracking issue
_https://github.com/flyteorg/flyte/issues/<number>_

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,19 +10,25 @@ ENV PYTHONPATH /root
 ARG VERSION
 ARG DOCKER_IMAGE
 
-RUN apt-get update && apt-get install build-essential -y
+# Combining RUN commands with the && : pattern at the end
+# 1. Update the necessary packages for flytekit
+# 2. Delete apt cache. Reference: https://gist.github.com/marvell/7c812736565928e602c4
+# 3. Change the permission of /tmp, so that others can run command on it
+RUN apt-get update && apt-get install build-essential libmagic1 -y \
```
Comment (eapolinario) [inline Dockerfile:17]: let's not install `libmagic1` on the docker image.

Comment (Future-Outlier) [reply Dockerfile:17]: no problem!

Comment (eapolinario) [reply Dockerfile]: The feature added in https://github.com/flyteorg/flytekit/pull/1893/ should be opt-in. We should make the default flytekit image as small as possible.

Comment (Future-Outlier) [reply Dockerfile]: Thanks a lot!

File: Dockerfile [hunk 2]
```diff
@@ -10,19 +10,25 @@ ENV PYTHONPATH /root
...
+        scikit-learn \
+    && apt-get clean autoclean \
+    && apt-get autoremove --yes \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/ \
+    && useradd -u 1000 flytekit \
+    && chown flytekit: /root \
+    && chown flytekit: /home \
+    && chown -R flytekit:flytekit /tmp \
+    && chmod 755 /tmp \
+    && :
```
[the remark this answers is not in the document]
Comment (Future-Outlier) [reply Dockerfile:30]: No problem!

**Review by eapolinario — APPROVED** [review]
Thank you!

**Pull request merged**